storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -430,12 +457,44 @@ public SupervisorDetails getSupervisorById(String nodeId) {
     }
 
     /**
+     * set assignments for cluster
+     */
+    public void setAssignments(Map<String, SchedulerAssignment> newAssignments) {
+        this.assignments = new HashMap<String, SchedulerAssignmentImpl>();
+        for (Map.Entry<String, SchedulerAssignment> entry : newAssignments.entrySet()) {
+            this.assignments.put(entry.getKey(), new SchedulerAssignmentImpl(entry.getValue().getTopologyId(), entry.getValue().getExecutorToSlot()));
```

Comment:
we could but the constructor kind of already takes care of that.  So the copyOf function will essentially be a wrapper for the SchedulerAssignmentImpl constructor